azerothcore/azerothcore-wotlk
# fix(Scripts/SlavePens): Rewrite Ahune.

Cherry-picked primarily from TC commits https://github.com/TrinityCore/TrinityCore/commit/e336272ec29dc5a4cc271205e2238d4087150b9c and https://github.com/TrinityCore/TrinityCore/commit/f507737ecff5d564da999c56d672274387d70ec2.



## Changes Proposed:

This PR proposes changes to:

## Issues Addressed:

- Closes unreported issues.

## SOURCE:

The changes have been validated through:

## Tests Performed:

This PR has been:
- [X] Lightly tested in-game by the author.

## How to Test the Changes:

1. `.event start 1`, `.debug lfg` if necessary.
2. Queue for the Frost Lord Ahune through random dungeon finder.
3. Engage the boss.
4. Observe his behavior.

## Known Issues and TODO List:

- [ ] The bunnies used for the shamans finding the opening (I believe that's the term) do not have movement. They need movement. (Would like to verify with sniffs)
- [ ] Script the stand spell and possibly a submerge equivalent to properly set up stand states. (Need to verify with sniffs)
- [ ] Bounding radii and combat reaches may need correction. (Need sniffs to correct)
- [ ] Loot from the piles of snow created by Ahune need creation.
- [ ] There is supposed to be an ice spot surrounding Ahune and his core. (Need sniffs to correct)
- [ ] The rhyming aura sometimes does not either land on the invoker or properly make the invoker start spitting bars.

The issues that require sniff data to be completed are not able to be completed currently due to a lack of 3.4.2 parsing, the only build Ahune was sniffed with (that I have). In my opinion, ~~if the Frozen Core is reachable by melee characters, then~~ this PR is an improvement from current behavior by a long shot and is okay to be merged with corrections being made down the line once more comprehensive sniffed data is available.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
